bin/install_cocoapods

```diff
@@ -13,13 +13,26 @@ LOCAL_CACHE_KEY="$BUILDKITE_PIPELINE_SLUG-local-pod-cache-$PODFILE_HASH"
 restore_cache "$LOCAL_CACHE_KEY"
 
 # If the `pod check` plugin is installed, use it to determine whether or not to install Pods at all
-# If it's not installed (or if it fails), we'll try to install Pods. 
+# If it's not installed (or if it fails), we'll try to install Pods.
 # If that fails, it may be due to an out-of-date repo. We can use `--repo-update` to try to resolve this automatically.
 if bundle exec pod plugins installed | grep -q check; then
 	bundle exec pod check || bundle exec pod install || bundle exec pod install --repo-update --verbose
 else
 	bundle exec pod install || bundle exec pod install --repo-update --verbose
 fi
 
+# Check that `Podfile.lock` was unchanged by `pod install`. If it was, it means
+# the lockfile might have been inadvertently changed.
+function lockfile_error () {
+  echo "Podfile.lock was changed by 'pod install'. Please run 'pod install' and try again."
```

Comment:
My test build is giving us a false positive. It fails ecause of `sha256sum`, so it does not verify the actual `Pofile.lock` check logic. Still, the error printing behavior is correct.

<img width="1118" alt="image" src="https://user-images.githubusercontent.com/1218433/160046280-49fc23d8-a8ce-491e-8856-bf4fc65215d8.png">

Looking at it in the builds logs, I think we ought to make it more visible to developers. Ideas:

1. Colorize it
2. Annotate the build